### DISCUSSION TITLE
Creating a Camera2D returns a Camera3D

### DISCUSSION THREAD

**Discussion by Caresle** [foreignsasquatch/raylib-hx#12] [Q&A] [answered]
Hi, i'm trying to create a Camera2D, but it looks like i get a Camera3D instead.

Code:
```haxe
var camera = Camera2D.create();
// Vector2 should be Vector3
camera.target = Vector2.create(0, 0);
// Camera has no field zoom
camera.zoom = 2;
```

And here are a list of things that i did to try to fix this:
- I did the 3d camera example from raylib and works
- I tried to cast Camera2D.create to Camera2D and doesn't work
- I tried to use the camera target with Vector3 and then add to a BeginMode2D(), throw "Camera should be Camera2D"

Example of cast camera2D:
```haxe
// Doesn't compile
var camera = cast(Camera2D.create(), Camera2D);
```

Example of use camera with vector3 and add to a BeginMode2D()
```haxe
var camera = Camera2D.create();
camera.target = Vector3.create(0, 0, 0);

while(!WindowShouldClose())
{
    BeginDrawing();
        BeginMode2D(camera);
            ClearBackground(Colors.RAYWHITE);
        EndMode2D();
    EndDrawing();
}
```

I'm pretty new to the haxe language, so i don't if i'm doing something bad.

**Comment by foreignsasquatch** [comment, marked as answer]
Thank you for noticing this 😓it seems that `Camera2D.create`  just returns a Camera instead of a Camera2D. It will be fixed soon :)

**Reply by foreignsasquatch** [reply]
remember to update your lib in haxelib via `haxelib git hxRaylib https://github.com/ForeignSasquatch/hxRaylib` as well

**Reply by Caresle** [reply]
Thank you for the answer, also thanks for the tip in how to update the lib :heart:
